src/broker/broker_admin_pub.c

```diff
@@ -322,8 +323,14 @@ admin_start_cmd (T_BROKER_INFO * br_info, int br_num, int master_shm_id, bool ac
     }
   chdir (envvar_bindir_file (path, BROKER_PATH_MAX, ""));
 
+  /* cannot execute broker unless find host name */
+  if (get_host_ip (ip_addr) < 0)
+    {
+      return -1;
+    }
+
```

Comment:
To minimize modifications, I propose the alternative:

1. add 
`#include "tcp.h"`
under 
#include "porting.h
2. add following in the decl. part.
`char hostname [CUB_MAXHOSTNAMELEN];`
4. replace get_host_ip () to the following
```
  if (GETHOSTNAME (hostname, CUB_MAXHOSTNAMELEN) != 0)
    {
      return -1;
    }

```

Reply:
I see. I will be better. Thank you.